Mathlib/Data/Nat/Bitwise.lean

```diff
@@ -44,6 +45,116 @@ namespace Nat
 
 set_option linter.deprecated false
 
+section
+variable {f : Bool → Bool → Bool}
+
+lemma bitwise'_bit' {f : Bool → Bool → Bool} (a : Bool) (m : Nat) (b : Bool) (n : Nat)
+    (ham : m = 0 → a = true) (hbn : n = 0 → b = true) :
+    bitwise' f (bit a m) (bit b n) = bit (f a b) (bitwise' f m n) := by
+  rw [bitwise', binaryRec_eq', binaryRec_eq']
+  · apply Or.inr hbn
+  · apply Or.inr ham
+
+@[simp]
+lemma bitwise_zero_left {m : Nat} : bitwise f 0 m = if f false true then m else 0 := by
+  rfl
```

Comment:
I default to `by rfl` because it's [supposedly faster](https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/What.20is.20the.20elaborator.20doing.20in.20this.20example) than `rfl` as a term, but the latter would also work.

Reply:
You should prefer `rfl` unless you have evidence that the speed matters (ie, the file becomes painful to edit)